### PR TITLE
chore: torch and pytorch-lightning as optional dependencies

### LINKED ISSUE #210: Move torch and pytorch-lightning packages to optional group
We want to disable the automatic torch installation since it's bloated and not the core functionality of this library.

### PR DESCRIPTION
**Pull request by Calychas**
Closes #210

### CODE REVIEW

File: srai/models/gtfs2vec_model.py [hunk 1]
```diff
@@ -41,29 +51,33 @@ def __init__(
...
-    def training_step(self, batch: torch.Tensor, batch_idx: Any) -> torch.Tensor:
+    def training_step(self, batch: "torch.Tensor", batch_idx: Any) -> "torch.Tensor":
         """
         Training step.
 
         Args:
             batch (torch.Tensor): Batch.
             batch_idx (Any): Batch index.
         """
+        from torch.nn import functional as F
```
Comment (piotrgramacki) [inline srai/models/gtfs2vec_model.py:79]: Will this import run only once? If not, maybe having it in a training step will hurt performance.

Comment (RaczeQ) [reply srai/models/gtfs2vec_model.py:79]: Based on this answer https://stackoverflow.com/a/37077911/7766101 it should run only once and not add significant overhead time.

Comment (piotrgramacki) [reply srai/models/gtfs2vec_model.py:79]: Ok, fine then ;)

**Review by piotrgramacki — DISMISSED** [review]
Nice work!

**Review by piotrgramacki — APPROVED** [review]

**Review by piotrgramacki — APPROVED** [review]

**Pull request merged**